torch_xla/csrc/lowering_context.cpp

```diff
@@ -52,24 +56,90 @@ class HloMetadataSetter {
         absl::StrReplaceAll(node->op().ToString(), {{":", "_"}});
     metadata.set_op_type(op_type);
     const torch::lazy::MetaData& nmeta = node->metadata();
-    std::string op_name_prefix;
-    if (!nmeta.scope.empty()) {
-      op_name_prefix =
-          absl::StrCat(absl::StrReplaceAll(nmeta.scope, {{":", "_"}}), "/");
+
+    ExtendedFrameInfo* efi = nullptr;
+
+    if (FLAGS_torch_lazy_ir_debug) {
+      TF_VLOG(1) << "PopulateXlaOpMetadata: Debug lowering enabled";
+      efi = static_cast<ExtendedFrameInfo*>(node->user_metadata());
+
+      if (efi != nullptr && efi->frames.size() != nmeta.frame_info.size()) {
+        LOG(WARNING) << "PopulateXlaOpMetadata: Extra frame information length "
+                        "does not match source "
+                     << "location length as expected " << efi->frames.size()
+                     << " != " << nmeta.frame_info.size();
+      }
+    } else {
+      TF_VLOG(1) << "PopulateXlaOpMetadata: Debug lowering is *not* enabled";
+    }
+
+    // Add class information iff there are frames
+    std::string last_class_name;
```

Comment:
Can the bit of code that computes frame be moved into a separate function?

Reply:
Taking a look to refactor